tests/unit/commands/init/test_cli.py

```diff
@@ -1387,3 +1408,29 @@ def test_init_cli_image_pool_with_base_image_having_one_managed_template_with_pr
                 no_input=None,
                 extra_context=None,
             )
+
+    @patch("samcli.lib.utils.git_repo.GitRepo._ensure_clone_directory_exists")
+    def test_init_cli_with_mismatch_dep_runtime(self, cd_mock):
```

Comment:
There also, check the exception type and message instead of only an `assertTrue`

Reply:
My bad, on the message, type is not accessible